pkg/resources/provisioning.cattle.io/v1/cluster/mutator.go

```diff
@@ -210,9 +261,10 @@ func (m *ProvisioningClusterMutator) handlePSACT(request *admission.Request, clu
 			// drop relevant fields if they exist in the cluster
 			dropMachineSelectorFile(machineSelectorFileForPSA(secretName, mountPath, ""), cluster, true)
 			args := getKubeAPIServerArg(cluster)
-			if args[kubeAPIAdmissionConfigOption] == mountPath {
-				delete(args, kubeAPIAdmissionConfigOption)
-				setKubeAPIServerArg(args, cluster)
+			if args.deleteByPredicate(func(arg keyValueArg) bool {
+				return arg.key == kubeAPIAdmissionConfigOption && arg.value == mountPath
+			}) {
+				setKubeAPIServerArg(*args, cluster)
```

Comment:
We should invoke the `setKubeAPIServerArg` function regardless of whether a specific argument is being deleted, in order to maintain consistent behavior.